### PR TITLE
An expired CRL should not override a successful match in other CRL

### PR DESCRIPTION
**Pull request by per-allansson**
# Description

If we have an expired CRL and loads another non-expired CRL, the current CRL check ends up returning the error from checking the CRL next date, while it found a good CRL match before that. This can be seen happening in wolfSSL debug logs here for example:

```
[2024-04-26T05:45:57.017Z] Info : [Gothenburg 1] Entering CheckCertCRL
[2024-04-26T05:45:57.017Z] Info : [Gothenburg 1] Found CRL Entry on list
[2024-04-26T05:45:57.017Z] Info : [Gothenburg 1] Checking next date validity
[2024-04-26T05:45:57.018Z] Info : [Gothenburg 1] Found CRL Entry on list
[2024-04-26T05:45:57.018Z] Info : [Gothenburg 1] Checking next date validity
[2024-04-26T05:45:57.018Z] Info : [Gothenburg 1] Date AFTER check failed
[2024-04-26T05:45:57.018Z] Info : [Gothenburg 1] CRL next date is no longer valid
[2024-04-26T05:45:57.018Z] Info : [Gothenburg 1]        CRL check not ok
 ...
[2024-04-26T05:45:57.018Z] Info : [Gothenburg 1] verify_callback err: -151
```

# Testing

Verified in internal tests.

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Comment by dgarske** [comment]
Okay to test. Contributor agreement on file.

File: src/crl.c [hunk 1]
```diff
@@ -426,16 +428,18 @@ static int CheckCertCRLList(WOLFSSL_CRL* crl, byte* issuerHash, byte* serial,
             #if !defined(NO_ASN_TIME) && !defined(WOLFSSL_NO_CRL_DATE_CHECK)
                 if (!XVALIDATE_DATE(crle->nextDate,crle->nextDateFormat, AFTER)) {
                     WOLFSSL_MSG("CRL next date is no longer valid");
-                    ret = ASN_AFTER_DATE_E;
+                    nextDateValid = 0;
                 }
             #endif
             }
-            if (ret == 0) {
+            if (nextDateValid) {
```
Comment (SparkiDev) [inline src/crl.c:435]: I don't see the difference made with this code change.
Please step it through for me.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: src/crl.c (hunk 1 above)
Comment (per-allansson) [reply src/crl.c:435]: You're right, it's not different at all, I missed the extra check below that it should not set ASN_AFTER_DATE_E if it already has a found entry. PR updated.

File: src/crl.c [hunk 2]
```diff
@@ -426,16 +428,18 @@ static int CheckCertCRLList(WOLFSSL_CRL* crl, byte* issuerHash, byte* serial,
...
                 foundEntry = 1;
                 ret = FindRevokedSerial(crle->certs, serial, serialSz,
                         serialHash, crle->totalCerts);
                 if (ret != 0)
                     break;
+            } else if (foundEntry == 0) {
```
Comment (SparkiDev) [inline src/crl.c:441]: Please make formatting consistent with rest of file:
```
} else if (...) {
->
}
else if (...) {
```

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: src/crl.c (hunk 2 above)
Comment (per-allansson) [reply src/crl.c]: fixed

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**